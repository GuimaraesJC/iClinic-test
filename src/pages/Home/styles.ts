import styled from "styled-components";

export const Container = styled.main`
height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  color: #2c97d1;
`;

export const Title = styled.h1`
  font-weight: normal;
  font-size: 72px;
  line-height: 88px;
  margin-top: 210px;

  span {
    font-family: 'Lato', sans-serif;
    font-weight: bold;
  }
`;

export const Subtitle = styled.h2`
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 17px;
  letter-spacing: 0.35em;
  text-transform: uppercase;
  margin-bottom: 162px;

  span {
    font-weight: bold;
  }
`;
