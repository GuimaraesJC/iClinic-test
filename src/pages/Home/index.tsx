import * as S from './styles';

function Home() {
  return (
    <S.Container>
      <S.Title>Welcome to <span>iClinic</span></S.Title>
      <S.Subtitle>Frontend challenge - by <span>Jean Carlos Guimar√£es</span></S.Subtitle>

      <S.Button>Start</S.Button>
    </S.Container>
  );
}

export default Home;
