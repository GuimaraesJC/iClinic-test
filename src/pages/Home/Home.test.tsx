import { render, screen } from '@testing-library/react';

import { BrowserRouter } from 'react-router-dom';


import Home from '.';

describe('<Home />', () => {
  it('should have a button with label "START"', () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );
    const buttonElement = screen.getByText(/start/i);
    expect(buttonElement).toBeInTheDocument();
  });

  it(`should contain the creator's name`, () => {
    render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );

    const subtitleElement = screen.getByRole('heading', { level: 2 });
    expect(subtitleElement).toContainHTML('<span>Jean Carlos Guimar√£es</span>');
  });

  it('should match the snapshot', () => {
    const { container } = render(
      <BrowserRouter>
        <Home />
      </BrowserRouter>
    );

    expect(container).toMatchSnapshot();
  });
});
