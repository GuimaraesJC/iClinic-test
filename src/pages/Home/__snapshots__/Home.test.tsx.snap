// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match the snapshot 1`] = `
.c3 {
  width: 189px;
  height: 56px;
  background: #2c97d1;
  border: none;
  border-radius: 10px;
  color: #fff;
  font-weight: bold;
  font-size: 18px;
  line-height: 22px;
  -webkit-letter-spacing: 0.35em;
  -moz-letter-spacing: 0.35em;
  -ms-letter-spacing: 0.35em;
  letter-spacing: 0.35em;
}

.c0 {
  height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #2c97d1;
}

.c1 {
  font-weight: normal;
  font-size: 72px;
  line-height: 88px;
  margin-top: 210px;
}

.c1 span {
  font-family: 'Lato',sans-serif;
  font-weight: bold;
}

.c2 {
  font-style: normal;
  font-weight: normal;
  font-size: 14px;
  line-height: 17px;
  -webkit-letter-spacing: 0.35em;
  -moz-letter-spacing: 0.35em;
  -ms-letter-spacing: 0.35em;
  letter-spacing: 0.35em;
  text-transform: uppercase;
  margin-bottom: 162px;
}

.c2 span {
  font-weight: bold;
}

<div>
  <main
    class="c0"
  >
    <h1
      class="c1"
    >
      Welcome to 
      <span>
        iClinic
      </span>
    </h1>
    <h2
      class="c2"
    >
      Frontend challenge - by 
      <span>
        Jean Carlos Guimar√£es
      </span>
    </h2>
    <a
      href="/master"
    >
      <button
        class="c3"
        label="START"
      >
        START
      </button>
    </a>
  </main>
</div>
`;
